src/components/dialog/dialog.test.tsx

```diff
@@ -0,0 +1,376 @@
+      maxHeight: "calc(100vh - 20px)",
+    });
+  });
+
+  it("sets correct data-* props on main dialog element", () => {
+    render(
+      <Dialog open title="My dialog" data-element="foo" data-role="bar" />,
+      { wrapper: AllTheProviders }
+    );
+    const dialog = screen.getByRole("dialog", { name: /My dialog/i });
+    expect(dialog).toHaveAttribute("data-component", "dialog");
+    expect(dialog).toHaveAttribute("data-element", "foo");
+    expect(dialog).toHaveAttribute("data-role", "bar");
+  });
+
+  it("sets custom data-* props on close button, when they are passed via the closeButtonDataProps prop", () => {
+    render(
+      <Dialog
+        open
+        onCancel={() => {}}
+        closeButtonDataProps={{
+          "data-element": "foo",
+          "data-role": "bar",
+        }}
+      />
+    );
+    const closeButton = screen.getByRole("button", { name: /Close/i });
+    expect(closeButton).toHaveAttribute("data-element", "foo");
+    expect(closeButton).toHaveAttribute("data-role", "bar");
+  });
+
+  it("renders with grey background when greyBackground prop is passed", () => {
+    render(<Dialog open title="My dialog" greyBackground />, {
+      wrapper: AllTheProviders,
+    });
+    expect(screen.getByRole("dialog")).toHaveStyle({
+      backgroundColor: "var(--colorsUtilityMajor025)",
+    });
+  });
+
+  it("does not apply vertical overflow styling to the content container when it contains a Form with a sticky footer", () => {
+    render(
+      <Dialog open>
+        <Form stickyFooter />
+      </Dialog>
+    );
+
+    const content = screen.getByTestId("dialog-content");
+    expect(content).not.toHaveStyle("overflow-y: auto");
+    expect(content).not.toHaveStyle("overflow-y: scroll");
+  });
+
+  it("is contained within a wrapper, which has the correct class names set, when className prop is passed", () => {
+    render(<Dialog open title="My dialog" className="special-dialog" />, {
+      wrapper: AllTheProviders,
+    });
+
+    const modalWrapper = screen.getByTestId("modal");
+    const dialog = within(modalWrapper).getByRole("dialog", {
+      name: /My dialog/i,
+    });
+
+    expect(dialog).toBeInTheDocument();
+    expect(modalWrapper).toHaveClass("carbon-dialog special-dialog");
+  });
+
+  it("does not position itself any closer than 20px from the top of the viewport", () => {
+    const originalInnerHeight = window.innerHeight;
+    window.innerHeight = 300;
+    jest
+      .spyOn(Element.prototype, "getBoundingClientRect")
+      .mockReturnValue({ height: 361 } as DOMRect);
+
+    render(<Dialog open title="My dialog" />, {
+      wrapper: AllTheProviders,
+    });
+    const dialog = screen.getByRole("dialog");
+
+    expect(dialog).toHaveStyle({ top: "20px" });
+
+    window.innerHeight = originalInnerHeight;
+    jest.restoreAllMocks();
+  });
+
+  it("does not position itself such that it goes off the left edge of the viewport", () => {
```

Comment:
question: I assume these tests (line 328 to 362) are needed for coverage? If not I think we're better letting chromatic and playwright test this stuff

Reply:
Yeah this is for coverage mainly 👍🏼